snafu/registry.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Implement a global tool registry that automatically maps tool names to their classes.
+
+This works by having a metaclass (added onto ```wrapper.Wrapper``), add tool classes to into the
+registry dict when the class is created. For more information on metaclasses, see:
+https://www.geeksforgeeks.org/python-metaclasses/.
+
+After snafu loads up and all the classes are created, the registry dict can be accessed at
+``registry.TOOLS``. Key names will be tool names, values will be their wrapper classes.
+"""
+from typing import Dict
+from abc import ABCMeta
+
+
+TOOLS: Dict[str, object] = dict()
+
+
+class ToolRegistryMeta(ABCMeta):
```

Comment:
This class is meant to replace the wrapper_factory by automating benchmark discovery: https://github.com/cloud-bulldozer/benchmark-wrapper/blob/master/snafu/utils/wrapper_factory.py